tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -461,8 +464,41 @@ def _py_all(iterable):
   return all(iterable)
 
 
+def sorted_(iterable, key=UNSPECIFIED, reverse=UNSPECIFIED):
+  if tensor_util.is_tensor(iterable):
+    return _tf_sorted(iterable, key, reverse)
+  return _py_sorted(iterable, key, reverse)
+
+
+def _tf_sorted(iterable, key, reverse):
+  """Overload of sorted_ for Tensor iterable."""
+  if reverse is UNSPECIFIED:
+    direction = 'ASCENDING'
+  else:
+    direction = 'DESCENDING'
+  if key is not UNSPECIFIED:
+    mapped = parallel_ops.vectorized_map(key, iterable)
+    with ops.control_dependencies(
```

Comment:
Optional: If the rank is static, you could add an extra check for an early warning:

```
if mapped.shape.rank is not None and mapped.shape.rank != 1:
  raise ValueError('sort only supports only 1D tensors')
```

Reply:
Thanks @mdanatg, the PR has been updated.